test/gtest/uct/uct_test.cc

```diff
@@ -1494,24 +1494,27 @@ void uct_test::async_event_ctx::signal() {
 }
 
 bool uct_test::async_event_ctx::wait_for_event(entity &e, double timeout_sec) {
-    if (wakeup_fd.fd == -1) {
+    /* Caching fd locally to workaroud bug in gcc (GCC) 11.2.1 20210728 FC35
+     * See details in https://github.com/openucx/ucx/issues/7705 */
+    struct pollfd local_fd = wakeup_fd;
+
+    if (local_fd.fd == -1) {
         /* create wakeup */
         if (e.iface_attr().cap.event_flags & UCT_IFACE_FLAG_EVENT_FD) {
-            ucs_status_t status =
-                uct_iface_event_fd_get(e.iface(), &wakeup_fd.fd);
+            ucs_status_t status = uct_iface_event_fd_get(e.iface(),
+                                                         &local_fd.fd);
             ASSERT_UCS_OK(status);
         } else {
-            ucs_status_t status =
-                ucs_async_pipe_create(&aux_pipe);
+            ucs_status_t status = ucs_async_pipe_create(&aux_pipe);
             ASSERT_UCS_OK(status);
             aux_pipe_init = true;
-            wakeup_fd.fd = ucs_async_pipe_rfd(&aux_pipe);
+            local_fd.fd   = ucs_async_pipe_rfd(&aux_pipe);
```

Comment:
@shamisp this causes fd leaks and test failures
need to set wakeup_fd here. and copy to local_fd right before calling poll()